Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -454,24 +454,32 @@ private static void MergeDictionary(IList<Dictionary<string, FlexMatch>> mergeFr
 
         private void RunMatchExpression(FlexMatch binary64DecodedMatch, AnalyzeContext context, MatchExpression matchExpression)
         {
+            bool isMalformed = true;
```

Comment:
>isMalformed [](http://example.com/codeflow?start=17&length=11)

this change will enable a rule to have:
- ContentsRegex
- IntrafileRegexes
- SinglelineRegexes

the idea is to re-use the match expression instead of duplicating it.